lib/Dialect/AIEVec/Transforms/AIEVectorize.cpp

```diff
@@ -2344,6 +2445,20 @@ void AIEVectorize::runOnOperation() {
     VectState *state =
         new VectState(func.getContext(), shiftParam, zeroOffset, dupFactor);
 
+    // First compute the loops surrounding each load/store operation. This is
+    // necessary to identify loads/stores that are nested together.
+    for (AffineForOp forOp : func.getOps<AffineForOp>()) {
+      SmallVector<Operation *, 8> enclosingLoops;
+      enclosingLoops.push_back(forOp);
+      computeEnclosingLoopsPerBlock(forOp, state, enclosingLoops);
+    }
+
+    // Check whether there is any unalignment loads.
+    if (unalignedLoadsCheck && failed(hasUnalignedLoads(func, state))) {
+      func.emitError("Unaligned loads\n");
```

Comment:
See my suggestion for 2313-2317. 
Perhaps change 2458 to func.emitError() << "Cannot apply aie-vectorize to " 
<< func->getName() << " because alignment check has failed.\n";